### DISCUSSION TITLE
Who feels responsible for the User-Interface-Guide?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-platform/.github#80] [General]
There is a wiki-page here: https://wiki.eclipse.org/User_Interface_Guidelines

It has several issues:

1. some links are broken
2. "This document is a working draft. Version 2.1 is the base with 3.x updates indicated with "(3.x update)"" ... Last major revision: **November 2007**
3. Most images look outdated compared to today Eclipse IDE
4. It seems more appropriate to manage this as part of Github wiki/page/... e.g. as part of this or an own repository so it is easier for other people to participate.

I'll send the link to this discussion also to the [ui-best-practices-working-group mailing list](https://accounts.eclipse.org/mailing-list/ui-best-practices-working-group), but if anyone here has more insights please share them!

**Comment by akurtakov** [comment]
If it hasn't been updated in 15 years looks like no one. I even question how useful having such a page is if no one felt the need for it for so long.

**Reply by laeubi** [reply]
> I even question how useful having such a page is if no one felt the need for it for so long.

At least that is something one find if searching for "eclipse styleguide"... if it is really irrelevant, we should either delete or mark it outdated. But what are the alternative then? It seems there are still some valuable information and files so probably we should save them somewhere? In general I think it would be good to have some common concepts across Eclipse plugins.

**Reply by akurtakov** [reply]
In most such cases reduction is the best path forward. Remove all outdated information and broken links as a first step. After that it would be easier to plan ahead.

**Comment by laeubi** [comment]
Mailinglist seems dead:
```
[ui-best-practices-working-group-request@eclipse.org](mailto:ui-best-practices-working-group-request@eclipse.org)
    host mail.eclipse.org [198.41.30.200]
    SMTP error from remote mail server after RCPT TO:[<ui-best-practices-working-group-request@eclipse.org>](mailto:ui-best-practices-working-group-request@eclipse.org):
    550 5.1.1 [<ui-best-practices-working-group-request@eclipse.org>](mailto:ui-best-practices-working-group-request@eclipse.org):
    Recipient address rejected: User unknown in local recipient table
```

**Reply by laeubi** [reply]
Alright my fault, the right address is obviously [ui-best-practices-working-group@eclipse.org](mailto:ui-best-practices-working-group@eclipse.org)

**Comment by merks** [comment]
No one feeling responsible to update it doesn't make it irrelevant.  That being said, at this point, I expect everyone just does whatever they personally think is best without regard for the opinions of others.  The end result is not pretty:

https://bugs.eclipse.org/bugs/show_bug.cgi?id=483982

**Reply by laeubi** [reply]
That's why I'm asking here, how can we reclaim responsibilities, if possible update and further enhance this, and if anything is needed here I'd like to bring this to the attention of the IDE-WG. The background is that we got some funding for m2e to improve user-experience and I'd need some normative guidelines (or an official statement that there are not anything anymore) to help the designers creating stuff that fits well into the Eclipse Eco-System...

**Reply by azoitl** [reply]
@laeubi if these people are given the current guidlines would they also be able to report issues in them? So that we can try to get them fixed?

**Reply by laeubi** [reply]
Sure I think so, at laest that was the intention: Having a way to ask question, maybe update things and so on, because of this I think a Github Repository is great in this regard, and even more usfull if we publish the content automatically to some page (e.g. gh pages)... as long as there is someone to review proposed changes.

**Reply by azoitl** [reply]
This would be great. I see that there is more interest in usability and user experience for the Eclipse Platform. At least I got that from the talks and discussion at this year's Eclipse Con. Although I must admit one talk was from me :-)

**Reply by laeubi** [reply]
@vogella I think you recently bring up some concerns regarding dialogs, do you like to bring that issue up in the repository so the guide might be adjusted?

**Comment by mickaelistria** [comment]
I believe instead of looking at Eclipse-specific rules, it's usually better to look outside. There are many resources on the web recommending some API design, helping to choose the right widget at the right time, helping to decide "how" to bug users with problems they'd rather not face, how to properly label widgets...
I used to browse https://goodui.org/patterns/ which I find inspiring; but most OS have recommendation when it comes to UI, and those are usually consistent.
But the bottom-line of UX is to design and test with empathy, trying to forget your technical knowledge of the why and how, and just judge your production with the eyes of a plain basic users. If a basic user can get it, then it's good enough.
When it comes to funded work, I suggest that instead of investing in guidelines not many people would read nor obey, you start by coding UX how you feel is best and then get it reviewed by "basic" users and empower them to give feedback and just obey the feedback as much as possible. Getting feedback for 3 distinct users (or customers) will be IMO faster and more productive towards building a good enough UX than trying to write some generic rules first.

**Reply by iloveeclipse** [reply]
> I believe instead of looking at Eclipse-specific rules, it's usually better to look outside. There are many resources on the web recommending some API design

One important UX rule is that an application should have **consistent** user interface. In web you can find many interesting but **contradictory** designs, so the point of the guideline **at Eclipse** is to give **application specific** guidelines to follow.

I personally find that if someone wants a guide, the https://wiki.eclipse.org/User_Interface_Guidelines should be still considered as first source of rules. If something doesn't match, open a question, discuss & update, **it is a Wiki**.

Also note: there are many unmaintained guidelines in the world (like [this one](https://en.wikipedia.org/wiki/Bible)) that are still considered valid by millions of people. No updates for years doesn't mean they are useless or should be totally ignored.

**Reply by laeubi** [reply]
> If something doesn't match, open a question, discuss & update, it is a Wiki.

Well but where to ask and who is giving normative answers? For sure I can just update everything as I see fit but that's not the goal, that's why I ask here. For example the wiki-page is even marked as [Pages with broken file links](https://wiki.eclipse.org/Category:Pages_with_broken_file_links) so who should I ask for the missing files? Who can give guidance if there is a disagreement?

So that's the topic of this ... not if we might want to wait a few thousands of years an hope our wiki has become the new bible ...

**Reply by iloveeclipse** [reply]
> Well but where to ask and who is giving normative answers?
> Who can give guidance if there is a disagreement?

Community of contributors called "platform project". 
Ask question here and you will see.

**Comment by azoitl** [comment]
I think the information that Gunnar tried to move it to github for better maintenance was lost over time.  https://github.com/eclipse-architecture/ui-best-practices

**Reply by laeubi** [reply]
@azoitl great many thanks for the information, I see there are an issue open and some discussions going on, but its a few years now as well, so any chance to start it again?

**Reply by azoitl** [reply]
I hope so. For me these guidelines where a great starting point when starting to extend Eclipse with own stuff.

**Reply by guw** [reply]
It never took off. I was doing the infrastructure work to get things moving but nothing happened since then. I think the old info was created/maintained by a team of UX people working at IBM.
